File: Pelikula.API/Model/Prodaja/ProdajaInsertRequest.cs
using System;
using System.Collections.Generic;

namespace Pelikula.API.Model.Prodaja
{
    public class ProdajaInsertRequest
    {
        public int? KorisnikId { get; set; }
        public int? RezervacijaId { get; set; }
        public DateTime? Datum { get; set; }

        public ICollection<ProdajaArtikalInsertRequest> ProdajaArtikal { get; set; }
    }
}

File: Pelikula.MOBILE/lib/model/prodaja/prodaja_insert_request.dart
class ProdajaInsertRequest {
  int? korisnikId;
  int? rezervacijaId;
  DateTime? datum;

  List<dynamic>? prodajaArtikal;

  ProdajaInsertRequest({
    this.korisnikId,
    this.rezervacijaId,
    this.datum,
    this.prodajaArtikal,
  });

  factory ProdajaInsertRequest.fromJson(Map<String, dynamic> json) {
    return ProdajaInsertRequest(
      korisnikId: json['korisnikId'] != null ? json['korisnikId'] as int : null,
      rezervacijaId: json['rezervacijaId'] as int,
      datum: DateTime.tryParse(json['datum']),
      prodajaArtikal: json['prodajaArtikal'] != null
          ? json['prodajaArtikal'] as List
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        "korisnikId": korisnikId,
        "rezervacijaId": rezervacijaId,
        "datum": datum == null ? null : datum!.toIso8601String(),
        "prodajaArtikal": prodajaArtikal,
      };
}
